File: DarkOrbit 10.0/Net/netty/commands/ClanRelationModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class ClanRelationModule
    {
        public const short ID = 27378;

        public const short NONE = 0;
        public const short ALLIED = 1;
        public const short NON_AGGRESSION_PACT = 2;
        public const short AT_WAR = 3;

        public short type = 0;
        public ClanRelationModule(short type)
        {
            this.type = type;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeShort(type);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/ClanRelationModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

public class ClanRelationModule
        implements ServerCommand {

    public static short NONE = 0;
    public static short ALLIED = 1;
    public static short NON_AGGRESSION_PACT  = 2;
    public static short AT_WAR  = 3;

    public static int ID = 2061;

    public short type = 0;

    public ClanRelationModule(short pType) {
        this.type = pType;
    }

    public int getID() {
        return ID;
    }

    public int method_1005() {
        return 0;
    }

    public void write(DataOutputStream out) {
        try {
            out.writeShort(ID);
            this.writeInternal(out);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    protected void writeInternal(DataOutputStream out) {
        try {
            out.writeShort(this.type);
            out.writeShort(15770);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}